File: src/models/user-resource.ts
import type { Attachment } from "./attachment";
import type { Tab } from "./tab";
import type { TabLocation } from "./tab-location";

export type UserResource = Readonly<{
  id: string
  kind: number
  name: string
  className?: string
  establishmentName: string
  profilePicture: Attachment | null
  isDirector: boolean
  isDelegate: boolean
  isMemberCA: boolean
  tabs: Map<TabLocation, Tab>
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/UserResource.kt
package ink.literate.pawnote.models

data class UserResource(
    val id: String,
    val kind: Int,
    val name: String,
    val className: String?,
    val establishmentName: String,
    val profilePicture: Attachment? = null,
    val isDirector: Boolean,
    val isDelegate: Boolean,
    val isMemberCA: Boolean,
    val tabs: Map<TabLocation, Tab>
)
